src/veranda/raster/native/geotiff.py

```diff
@@ -0,0 +1,498 @@
+        """ list of str : Scale factors of the different bands. """
+        return list(self._scale_factors.values())
+
+    @property
+    def offsets(self):
+        """ list of numbers : Offsets of the different bands. """
+        return list(self._offsets.values())
+
+    @property
+    def nodatavals(self):
+        """ list of numbers : No data values of the different bands. """
+        return list(self._nodatavals.values())
+
+    @property
+    def color_interps(self):
+        """ list of numbers : Color interpretation values of the different bands. """
+        return list(self._color_intprs.values())
+
+    @property
+    def color_tables(self):
+        """ list : Color tables of the different bands. """
+        return list(self._color_tbls.values())
+
+    @property
+    def dtypes(self):
+        """ list of str : Data types in NumPy-style format. """
+        return [GDAL_TO_NUMPY_DTYPE[dtype] for dtype in self._dtypes.values()]
+
+    def _open(self):
+        """
+        Helper function supporting the different file modes, i.e. either opening an existing file or creating a new
+        file source.
+
+        """
+        if self.mode == 'r':
+            if not os.path.exists(self.filepath):
+                err_msg = f"File '{self.filepath}' does not exist."
+                raise FileNotFoundError(err_msg)
+            self.src = gdal.Open(self.filepath, gdal.GA_ReadOnly)
+            self.shape = self.src.RasterYSize, self.src.RasterXSize
+            self.geotrans = self.src.GetGeoTransform()
+            self.sref_wkt = self.src.GetProjection()
+            self.metadata = self.src.GetMetadata()
+            self.blocksize = self.src.GetRasterBand(1).GetBlockSize()  # block seems to be band-independent, because no set function is available per band
+            self.compression = self.src.GetMetadata('IMAGE_STRUCTURE').get('COMPRESSION')
+            self.is_bigtiff = self.is_file_bigtiff(self.filepath)
+            self.is_tiled = self.blocksize[1] == 1
+
+            self.bands = []
+            self._scale_factors = dict()
+            self._offsets = dict()
+            self._nodatavals = dict()
+            self._color_tbls = dict()
+            self._color_intprs = dict()
+            self._dtypes = dict()
+            for band in range(1, self.src.RasterCount + 1):
+                self.bands.append(band)
+                scale_factor = self.src.GetRasterBand(band).GetScale()
+                offset = self.src.GetRasterBand(band).GetOffset()
+                self._scale_factors[band] = scale_factor or 1
+                self._offsets[band] = offset or 0
+                self._nodatavals[band] = self.src.GetRasterBand(band).GetNoDataValue()
+                self._color_tbls[band] = self.src.GetRasterBand(band).GetColorTable()
+                self._color_intprs[band] = self.src.GetRasterBand(band).GetColorInterpretation()
+                self._dtypes[band] = self.src.GetRasterBand(band).DataType
```

Comment:
extract

Also looks like I've seen similar before

Reply:
Done. I am sure this can be improved, but for the time being I think its okay.